### PR TITLE
Add endpoint for syncing areas of interest with subscriptions

### PR DESCRIPTION
**Pull request by ikas**
This PR adds a new endpoint to allow syncing areas with subscriptions manually. Pagination was also added when fetching areas with all=true filter.

### CODE REVIEW

File: app/test/e2e/v2/get-area.spec.js [hunk 1]
```diff
@@ -103,9 +103,17 @@ describe('Get areas - V2', () => {
             [{ userId: USERS.USER.id }, { userId: USERS.MANAGER.id }, { userId: USERS.ADMIN.id }]
         );
 
-        const response = await requester.get(`/api/v2/area?loggedUser=${JSON.stringify(USERS.ADMIN)}&all=true`);
+        // First sync areas
+        const response = await requester.post(`/api/v2/area/sync`).send({ loggedUser: USERS.ADMIN });
```
Comment (tiagojsag) [inline app/test/e2e/v2/get-area.spec.js]: this is basically a single test for two endpoints - not sure I like it.

File: app/test/e2e/v2/get-area.spec.js [hunk 2]
```diff
@@ -103,9 +103,17 @@ describe('Get areas - V2', () => {
...
         response.status.should.equal(200);
-        response.body.should.have.property('data').and.be.an('array').and.have.length(5);
+        response.body.should.have.property('data').and.be.an('object');
+        response.body.data.should.have.property('syncedAreas').and.equal(1);
+        response.body.data.should.have.property('createdAreas').and.equal(2);
+
+        // Get all areas
+        const getResponse = await requester.get(`/api/v2/area?loggedUser=${JSON.stringify(USERS.ADMIN)}&all=true`);
+        getResponse.status.should.equal(200);
+        getResponse.body.should.have.property('data').and.be.an('array').and.have.length(5);
```
Comment (tiagojsag) [inline app/test/e2e/v2/get-area.spec.js]: this needs additional validations

File: app/test/e2e/v2/get-area.spec.js [hunk 3]
```diff
@@ -182,32 +197,69 @@ describe('Get areas - V2', () => {
...
+        const area4 = await new Area(createArea()).save();
+        const area5 = await new Area(createArea()).save();
+
+        const response = await requester.get(`/api/v2/area?loggedUser=${JSON.stringify(USERS.ADMIN)}&all=true&page[size]=3`);
+        response.status.should.equal(200);
+        response.body.should.have.property('data').with.lengthOf(3);
+        response.body.should.have.property('links').and.be.an('object');
+
+        const ids = response.body.data.map((area) => area.id);
+        ids.should.contain(area1.id);
```
Comment (tiagojsag) [inline app/test/e2e/v2/get-area.spec.js]: https://medium.com/building-ibotta/testing-arrays-and-objects-with-chai-js-4b372310fe6d

File: app/test/e2e/utils/helpers.js [hunk 4]
```diff
@@ -134,6 +134,7 @@ const mockSubscriptionFindForUser = (userId, idsList = []) => {
 const mockSubscriptionFindAll = (ids = [], overrideArray = {}) => {
     nock(process.env.CT_URL)
         .get(`/v1/subscriptions/find-all`)
+        .query(() => true)
```
Comment (tiagojsag) [inline app/test/e2e/utils/helpers.js]: this is evil - make the mocks match the actual requests.

File: app/test/e2e/v2/get-area.spec.js (hunk 1 above)
Comment (ikas) [reply app/test/e2e/v2/get-area.spec.js]: The sync call needs to be here to simulate a real use case (which is calling sync + get with all=true filter). This is a test for the get endpoint with all=true, and there's a separate test suite for the sync endpoint, so not testing both endpoints here :)

File: app/test/e2e/v2/get-area.spec.js (hunk 2 above)
Comment (ikas) [reply app/test/e2e/v2/get-area.spec.js]: Added assertions for the ids returned here!

File: app/test/e2e/v2/get-area.spec.js (hunk 3 above)
Comment (ikas) [reply app/test/e2e/v2/get-area.spec.js]: Thanks for the tip, quite nice :ok_hand:

File: app/test/e2e/utils/helpers.js (hunk 4 above)
Comment (ikas) [reply app/test/e2e/utils/helpers.js]: Validating if the correct params are being provided!

**Pull request merged**